guides/source/association_basics.md

```diff
@@ -140,210 +236,198 @@ create_table :books do |t|
+The `association=` method assigns an associated object to this object. Behind
+the scenes, this means extracting the primary key from the associated object and
+setting this object's foreign key to the same value.
 
 ```ruby
-class Physician < ApplicationRecord
-  has_many :appointments
-  has_many :patients, through: :appointments
-end
-
-class Appointment < ApplicationRecord
-  belongs_to :physician
-  belongs_to :patient
-end
-
-class Patient < ApplicationRecord
-  has_many :appointments
-  has_many :physicians, through: :appointments
-end
+@book.author = @author
 ```
 
-![has_many :through Association Diagram](images/association_basics/has_many_through.png)
 
-The corresponding migration might look like this:
+The `build_association` method returns a new object of the associated type. This
+object will be instantiated from the passed attributes, and the link through
+this object's foreign key will be set, but the associated object will _not_ yet
+be saved.
 
 ```ruby
-class CreateAppointments < ActiveRecord::Migration[7.2]
-  def change
-    create_table :physicians do |t|
-      t.string :name
-      t.timestamps
-    end
+@author = @book.build_author(author_number: 123,
+                             author_name: "John Doe")
+```
 
-    create_table :patients do |t|
-      t.string :name
-      t.timestamps
-    end
+The `create_association` method takes it a step further and also saves the
+associated object once it passes all of the validations specified on the
+associated model.
 
-    create_table :appointments do |t|
-      t.belongs_to :physician
-      t.belongs_to :patient
-      t.datetime :appointment_date
-      t.timestamps
-    end
-  end
-end
+```ruby
+@author = @book.create_author(author_number: 123,
+                              author_name: "John Doe")
 ```
 
-The collection of join models can be managed via the [`has_many` association methods](#has-many-association-reference).
-For example, if you assign:
+Finally, `create_association!` does the same, but raises
+`ActiveRecord::RecordInvalid` if the record is invalid.
 
 ```ruby
-physician.patients = patients
+# This will raise ActiveRecord::RecordInvalid because the name is blank
+begin
+  @book.create_author!(author_number: 123, name: "")
+rescue ActiveRecord::RecordInvalid => e
+  puts e.message
+end
+```
+
+```irb
+irb> raise_validation_error: Validation failed: Name can't be blank (ActiveRecord::RecordInvalid)
 ```
 
-Then new join models are automatically created for the newly associated objects.
-If some that existed previously are now missing, then their join rows are automatically deleted.
+##### Checking for Association Changes
```

Comment:
Do you think it's worth mentioning that `model.association_changed?` and `model.association.changed?` are different? The first one tracks if there's a new record, while the second one tracks if some attribute in the association changed. If we add this note, _maybe_ it is worth also mentioning that this is backed by `ActiveModel::Dirty`

Reply:
Sounds good! I've added a note to the bottom:

```
NOTE: Do not confuse `model.association_changed?` with
`model.association.changed?`. The former checks if the association has been
replaced with a new record, while the latter tracks changes to the attributes of
the association.
```

However, I've chosen not to mention `ActiveModel::Dirty` as I think its more relevant to the ActiveModel section and we don't need to go into details here.